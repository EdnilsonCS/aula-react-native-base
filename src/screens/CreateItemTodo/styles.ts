import styled from 'styled-components/native';
import IconFeather from 'react-native-vector-icons/Feather';

export const Container = styled.View`
  flex: 1;
  justify-content: center;
  padding: 40px;
`;

export const Icon = styled(IconFeather)`
  align-self: center;
`;

export const Title = styled.Text`
  font-size: 40px;
  align-self: center;
`;
