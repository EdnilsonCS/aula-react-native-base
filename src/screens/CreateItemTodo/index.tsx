import React from 'react';
import { TextInput, Button } from '@components/index';
import * as Yup from 'yup';
import { useForm } from 'react-hook-form';
import { yupResolver } from '@hookform/resolvers/yup';
import { useTodo } from '@hooks/todo';
import { useNavigation } from '@react-navigation/native';
import { Container, Icon, Title } from './styles';

interface TodoItem {
  title: string;
  description: string;
}

const Login: React.FC = () => {
  const { handleAddItem } = useTodo();
  const navigation = useNavigation();
  const schema = Yup.object().shape({
    description: Yup.string().required('Descrição é um campo obrigatório'),
    title: Yup.string().required('Titulo obrigatória'),
  });

  const { control, handleSubmit, errors } = useForm({
    resolver: yupResolver(schema),
    mode: 'onBlur',
    defaultValues: {
      email: '',
      title: '',
    },
  });
  const onSubmit = (data: TodoItem): void => {
    handleAddItem(data);
    navigation.navigate('CreateItemTodoSuccess');
  };
  return (
    <Container>
      <Icon name="plus-circle" size={200} />
      <Title>Criar um Novo Item</Title>
      <TextInput
        name="title"
        errors={errors}
        control={control}
        placeholder="Titulo"
        icon="edit-2"
        testID="title-input"
      />

      <TextInput
        name="description"
        errors={errors}
        control={control}
        placeholder="Descrição"
        icon="edit"
        testID="description-input"
      />

      <Button onPress={handleSubmit(onSubmit)}>Criar novo item</Button>
    </Container>
  );
};

export default Login;
