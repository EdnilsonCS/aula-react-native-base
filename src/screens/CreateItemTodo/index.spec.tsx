import React from 'react';
import { act, fireEvent, render } from '@testing-library/react-native';
import CreateItemTodo from './index';

const mockedHistoryPush = jest.fn();

jest.mock('@react-navigation/native', () => {
  return {
    useNavigation: () => ({
      goBack: jest.fn(),
      navigate: mockedHistoryPush,
    }),
  };
});

const mockedHandleAdd = jest.fn();
jest.mock('../../hooks/todo.tsx', () => {
  return {
    useTodo: () => ({
      handleAddItem: mockedHandleAdd,
    }),
  };
});

describe('CreateItemTodo', () => {
  it('Should be able to render correct', () => {
    const tree = render(<CreateItemTodo />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should be able to add a new item', async () => {
    const { getByPlaceholderText, getByText } = render(<CreateItemTodo />);
    const descriptionField = getByPlaceholderText('Descrição');
    const titleField = getByPlaceholderText('Titulo');
    const buttonElement = getByText('Criar novo item');

    await act(async () => {
      fireEvent.changeText(titleField, '');
      fireEvent.changeText(descriptionField, '');
      fireEvent.changeText(titleField, 'fake-title');
      fireEvent.changeText(descriptionField, 'fake-description');
      fireEvent.press(buttonElement);
    });

    expect(mockedHistoryPush).toHaveBeenCalledWith('CreateItemTodoSuccess');
    expect(mockedHandleAdd).toHaveBeenCalledWith({
      title: 'fake-title',
      description: 'fake-description',
    });
  });
});
