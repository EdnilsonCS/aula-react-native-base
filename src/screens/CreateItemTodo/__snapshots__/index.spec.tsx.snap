// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateItemTodo Should be able to render correct 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "paddingBottom": 40,
        "paddingLeft": 40,
        "paddingRight": 40,
        "paddingTop": 40,
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 200,
        },
        Array [
          Object {
            "alignSelf": "center",
          },
        ],
        Object {
          "fontFamily": "Feather",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
          "fontSize": 40,
        },
      ]
    }
  >
    Criar um Novo Item
  </Text>
  <View
    style={
      Object {
        "marginBottom": 10,
      }
    }
  >
    <View
      isErrored={false}
      isFocused={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFA800",
            "borderRadius": 10,
            "flexDirection": "row",
            "height": 60,
            "justifyContent": "space-between",
            "paddingBottom": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 0,
            "width": "100%",
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        keyboardAppearance="dark"
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Titulo"
        placeholderTextColor="rgba(0, 0, 0, 0.6)"
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
        value=""
      />
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#666360",
              "fontSize": 20,
            },
            Array [
              Object {
                "marginRight": 16,
              },
            ],
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginBottom": 10,
      }
    }
  >
    <View
      isErrored={false}
      isFocused={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFA800",
            "borderRadius": 10,
            "flexDirection": "row",
            "height": 60,
            "justifyContent": "space-between",
            "paddingBottom": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 0,
            "width": "100%",
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        keyboardAppearance="dark"
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Descrição"
        placeholderTextColor="rgba(0, 0, 0, 0.6)"
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
        value=""
      />
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#666360",
              "fontSize": 20,
            },
            Array [
              Object {
                "marginRight": 16,
              },
            ],
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1F1C1C",
        "borderRadius": 5,
        "justifyContent": "center",
        "minWidth": 96,
        "opacity": 1,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 14,
            "letterSpacing": 1.25,
            "lineHeight": 16,
          },
        ]
      }
    >
      Criar novo item
    </Text>
  </View>
</View>
`;
